staging/src/k8s.io/cli-runtime/pkg/genericclioptions/resource/builder.go

```diff
@@ -130,9 +130,27 @@ func IsUsageError(err error) bool {
 
 type FilenameOptions struct {
 	Filenames []string
+	Kustomize []string
 	Recursive bool
 }
 
+func (o *FilenameOptions) Validate() error {
+	if len(o.Filenames) > 0 && len(o.Kustomize) > 0 {
```

Comment:
it's likely many FilenameOptions/Builder users will not call this new method. should this validation go into `Builder#FilenameParam` and append to Builder#errs instead?

Reply:
I added two functions to `FilenameOptions`: `Validate()` and `IsSourceEmpty()`. 

`Validate()` is to make sure there is no `-f` and `-k` conflicts. This function is needed wherever `FilenameOptions` is used. So it makes sense to move it into `Builder#FilenameParam`.  

`IsSourceEmpty()` is to check both `-f` and `-k` are not set. This is used to make sure at least one of `-f` and `-k` is specified. There are subcommands such as `apply`. Previously, `-f` is required in those subcommands. So we need to change them to require one of `-f` and `-k`.

Reply:
`validate` now is triggered inside `FilenameParam` function. In addition, I renamed `IsSourceEmpty()` to `RequireFilenameOrKustomize`.